src/hdmf/common/table.py

```diff
@@ -42,6 +42,7 @@ class VectorData(Data):
              'doc': 'a dataset where the first dimension is a concatenation of multiple vectors', 'default': list()},
             allow_positional=AllowPositional.WARNING)
     def __init__(self, **kwargs):
+        kwargs['data'] = H5DataIO(data=kwargs['data'], maxshape=(None,))
```

Comment:
Hardcoding `H5DataIO` will probably create issues with `ZarrIO`. In the best case, `ZarrIO` ignores the wrapper and worst case it would fail.... I'm not sure which. For `ZarrIO` we should not need to wrap for datasets to be expandable, but the problem is, we won't know which backend will be used until we call write to know which wrapper to use.

Not sure what the best solution is for this. I could imagine a few approaches:

1.  Have "expandable" as an option on the base `DataIO` class and wrap with `DataIO` as a generic wrapper that the specific backend (HD5FIO, ZarrIO etc.) would then need to know how to translate. ---> I think this should be feasible, but would require changes in both `HDF5IO` and `ZarrIO`
2. Maybe we would need to wrap in the build process  --->  I think this could work, but may be tricky 
3. Wrap with `H5DataIO` and require that the other backends know how to translate it  ---> I don't like this one, because it hard-codes backend-specific wrappers in the Container

Reply:
This is what was going through my mind as well. I made a note documenting that. I had a expandable field idea.